wolfcrypt/src/asn.c

```diff
@@ -23321,24 +23324,26 @@ int wc_PubKeyPemToDer(const unsigned char* pem, int pemSz,
 
     WOLFSSL_ENTER("wc_PubKeyPemToDer");
 
-    if (pem == NULL || buff == NULL || buffSz <= 0) {
+    if (pem == NULL || (buff != NULL && buffSz <= 0)) {
         WOLFSSL_MSG("Bad pem der args");
         return BAD_FUNC_ARG;
     }
 
     ret = PemToDer(pem, pemSz, PUBLICKEY_TYPE, &der, NULL, NULL, NULL);
-    if (ret < 0 || der == NULL) {
+    if (ret == 0 && der != NULL && buff == NULL) {
```

Comment:
Same change here - check error first.

Reply:
Sorry, I should have caught that.  It is fixed now. .:)